src/screens/UserStats/index.tsx

```diff
@@ -0,0 +1,336 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { Alert } from 'react-native';
+import { Avatar, Fab, useTheme, useToast } from 'native-base';
+import storage from '@react-native-firebase/storage';
+import firestore, {
+  FirebaseFirestoreTypes,
+} from '@react-native-firebase/firestore';
+import {
+  useRoute,
+  useIsFocused,
+  useNavigation,
+} from '@react-navigation/native';
+import auth, { FirebaseAuthTypes } from '@react-native-firebase/auth';
+import { UserCirclePlus, UserCircleMinus } from 'phosphor-react-native';
+
+import Toast from '@components/Toast';
+import VStack from '@components/VStack';
+import ScrollView from '@components/ScrollView';
+import Loading from '@components/Loading';
+import Attribute from '@components/Attribute';
+import ScreenWrapper from '@components/ScreenWrapper';
+import genderTranslator from '@hashmaps/genderTranslator';
+import { ProfileDto } from '@interfaces/profile.dto';
+import { PartyDto } from '@interfaces/party.dto';
+import { useAppDispatch } from '@store/index';
+import { setTitle } from '@store/slices/navigation-slice';
+import { AXIS_X_PADDING_CONTENT, TOAST_DURATION } from '@utils/constants';
+import firestoreValueIsValid from '@utils/firestoreValueIsValid';
+
+type RouteParams = {
+  uuid: string;
+};
+
+const UserStats = () => {
+  const toast = useToast();
+  const { colors } = useTheme();
+  const navigation = useNavigation();
+  const dispatch = useAppDispatch();
+  const route = useRoute();
+  const isFocused = useIsFocused();
+  const { uuid } = route.params as RouteParams;
+  const [isMember, setIsMember] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [profile, setProfile] = useState<ProfileDto>({} as ProfileDto);
+  const [image, setImage] = useState<string | undefined>(undefined);
+  const userSession: FirebaseAuthTypes.User = auth().currentUser!;
+  const profileRef = useRef<
+    FirebaseFirestoreTypes.DocumentReference<ProfileDto>
+  >(firestore().collection<ProfileDto>('profiles').doc(uuid));
+  const partyRef = useRef<FirebaseFirestoreTypes.DocumentReference<PartyDto>>(
+    firestore().collection<PartyDto>('parties').doc(userSession.uid),
+  );
+
+  useEffect(() => {
+    let isMounted = true;
+
+    isMounted && isFocused && dispatch(setTitle('User Stats'));
+
+    return () => {
+      isMounted = false;
+    };
+  }, [isFocused]);
+
+  const checkMember = async () => {
+    try {
+      const response = (await partyRef.current.get()).data();
+
+      if (response !== undefined) {
+        setIsMember(response.members.includes(uuid));
+      }
```

Comment:
can we change response to null and well defined?

Reply:
The profileStats is already typed, undefined is a posible response of .data() method